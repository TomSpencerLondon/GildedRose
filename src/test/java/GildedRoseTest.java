import static org.approvaltests.Approvals.verifyAll;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.ArrayList;
import java.util.List;
import org.junit.jupiter.api.Test;


public class GildedRoseTest {

	@Test
	public void acceptanceTest() {
		List<Item> items = new ArrayList<>();
		items.add(new Item("+5 Dexterity Vest", 10, 20));
		items.add(new Item("Aged Brie", 2, 0));
		items.add(new Item("Elixir of the Mongoose", 5, 7));
		items.add(new Item("Sulfuras, Hand of Ragnaros", 0, 80));
		items.add(new Item("Backstage passes to a TAFKAL80ETC concert", 15, 20));
		items.add(new Item("Conjured Mana Cake", 3, 6));
		items.add(new Item("Elixir of the Mongoose", 1, -1));
		List<String> result = new ArrayList<>();
		GildedRose.updateQuality(items);
		for (Item item : items) {
			result.add(String.format("Name: %s, SellIn: %d, Quality: %d",
					item.getName(),
					item.getSellIn(),
					item.getQuality()));
		}

		verifyAll("", result);
	}
}
