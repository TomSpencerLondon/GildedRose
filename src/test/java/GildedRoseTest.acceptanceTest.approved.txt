[0] = Name: +5 Dexterity Vest, SellIn: 9, Quality: 19
[1] = Name: Aged Brie, SellIn: 1, Quality: 1
[2] = Name: Elixir of the Mongoose, SellIn: 4, Quality: 6
[3] = Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: 80
[4] = Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 14, Quality: 21
[5] = Name: Conjured Mana Cake, SellIn: 2, Quality: 5
[6] = Name: Elixir of the Mongoose, SellIn: 0, Quality: -1
[7] = Name: Aged Brie, SellIn: 0, Quality: 51
[8] = Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 9, Quality: 22
[9] = Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 9, Quality: 51
[10] = Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 9, Quality: 50
[11] = Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 4, Quality: 50
[12] = Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 4, Quality: 50
[13] = Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 4, Quality: 50
[14] = Name: +5 Dexterity Vest, SellIn: -2, Quality: 0
[15] = Name: +5 Dexterity Vest, SellIn: -2, Quality: 1
[16] = Name: Aged Brie, SellIn: -2, Quality: 5
[17] = Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0
[18] = Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: 3
[19] = Name: Aged Brie, SellIn: -2, Quality: 51
